scripts/usb_scan.py

```diff
@@ -0,0 +1,618 @@
+                        allow_interface = True
+                        break
+                    else:
+                        log_wrapper("deny " + i.get_node())
+                        return False
+            if not allow_interface:
+                log_wrapper("deny " + i.get_node() + ", no matching rule")
+                return False
+
+        log_wrapper("allow " + device.get_node())
+        return True
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="scanner to get USB devices info")
+    parser.add_argument("-d", "--diagnostic", dest="diagnostic",
+                        action="store_true",
+                        help="enable diagnostic mode")
+    return parser.parse_args()
+
+
+def to_pusb(device):
+    """ convert UsbDevice to pusb dict
+
+    Example pusb dict:
+    [{"product-desc": "", "product-id": "383c", "description":
+    "Lenovo_H8RU2TGE", "vendor-desc": "Lenovo", "version": "2.00", "vendor-id":
+    "17ef", "path": "1-2", "serial": "H8RU2TGE"}, {"product-desc":
+    "TransMemory-Mini / Kingston DataTraveler 2.0 Stick (2GB)",
+    "product-id": "6544", "description":"Toshiba Corp._TransMemory-Mini /
+    Kingston DataTraveler 2.0 Stick (2GB)_9875B1B3F9F4CD301917964B",
+      "vendor-desc": "Toshiba Corp.", "version": "2.00", "vendor-id": "0930",
+      "path": "4-1.2", "serial": "9875B1B3F9F4CD301917964B"}]
+
+    :param device:(UsbDevice) the device to convert
+    :return:(dict) the key value pairs for pusb
+    """
+    pusb = {}
+
+    pusb["path"] = device.get_node()
+    # strip the leading space of "version"
+    pusb["version"] = device[UsbDevice._VERSION].strip()
+    pusb["vendor-id"] = device[UsbDevice._ID_VENDOR]
+    pusb["product-id"] = device[UsbDevice._ID_PRODUCT]
+    pusb["vendor-desc"] = device[UsbDevice._DESC_VENDOR]
+    pusb["product-desc"] = device[UsbDevice._DESC_PRODUCT]
+    pusb["serial"] = device[UsbDevice._SERIAL]
+
+    pusb["description"] = pusb["vendor-desc"]
+    if pusb["product-desc"]:
+        pusb["description"] += "_" + pusb["product-desc"]
+    if pusb["serial"]:
+        pusb["description"] += "_" + pusb["serial"]
+
+    return pusb
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    if args.diagnostic:
+        __DIAGNOSTIC = True
+
+    # get usb info
+    devices, interfaces = get_usb_info()
+
+    # debug info
+    log_list_wrapper(devices)
+    log_list_wrapper(interfaces)
+
+    # match interface to device
+    for i in interfaces:
+        for d in devices:
+            if i.is_child_of(d):
+                d.add_interface(i)
+
+    log_list_wrapper(devices)
+
+    # do policy check
+    policy = Policy()
+    pusbs = []
+    for d in devices:
+        if d.is_ready():
+            d.set_allow(policy.check(d))
+            if d.is_allow():
+                pusbs.append(to_pusb(d))
+        else:
+            log_wrapper("ignore not ready device: " + str(d))
+
+    # pass pusbs in json to XAPI
+    print(json.dumps(pusbs))
```

Comment:
I'd like to see the expected JSON output for an example included in the documentation. This JSON should be well formatted such that it is easy to follow. Like this:
```
[
  {
    "product-desc": "",
    "product-id": "0401",
    "description": "Feitian Technologies, Inc.",
    "vendor-desc": "Feitian Technologies, Inc.",
    "version": "1.10",
    "vendor-id": "096e",
    "path": "1-2",
    "serial": ""
  },
  {
    "product-desc": "",
    "product-id": "1887",
    "description": "MediaTek Inc._KXLGCA04127",
    "vendor-desc": "MediaTek Inc.",
    "version": "2.00",
    "vendor-id": "0e8d",
    "path": "1-4",
    "serial": "KXLGCA04127"
  }
]

```

Reply:
done.  put it in the  comments of to_push function.